src/Proxy/PurchaseLedgerPack.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Linn.Stores.Proxy
+{
+    using System.Data;
+
+    using Linn.Common.Proxy.LinnApps;
+    using Linn.Stores.Domain.LinnApps.ExternalServices;
+
+    using Oracle.ManagedDataAccess.Client;
+
+    public class PurchaseLedgerPack : IPurchaseLedgerPack
+    {
+        private readonly IDatabaseService databaseService;
+
+        public PurchaseLedgerPack(IDatabaseService databaseService)
+        {
+            this.databaseService = databaseService;
+        }
+
+        public int GetLedgerPeriod()
+        {
+            using (var connection = this.databaseService.GetConnection())
+            {
+                connection.Open();
+                var cmd = new OracleCommand("pl_pack.get_pl_ledger_period", connection)
+                              {
+                                  CommandType = CommandType.StoredProcedure
+                              };
+                var result = new OracleParameter("document_type", OracleDbType.Varchar2)
+                                 {
+                                     Direction = ParameterDirection.ReturnValue, Size = 50
+                                 };
+                cmd.Parameters.Add(result);
+
+                cmd.ExecuteNonQuery();
+                connection.Close();
+                var res = result.Value.ToString();
+                return int.Parse(res);
+            }
+        }
+
+        public int GetNextLedgerSeq()
+        {
+            using (var connection = this.databaseService.GetConnection())
+            {
+                connection.Open();
+                var cmd = new OracleCommand("cg_code_controls_next_val", connection)
```

Comment:
This is already in the common database service, but I need to get multiple values from it with my domain service (which can't use the common db service) to insert into purchase ledger. Felt it was better to recreate here and just call from domain, than to put extra logic in facade and pass in multiple values

Reply:
you can do it in the repository

Reply:
Or should I have a purchaseLedgerFacadeService + call add for each detail being posted, and just do the normal databaseService "get next val" call within the createFromResource of the purchaseLedgerFacadeService?

Reply:
Ahh will give that a go! Sorry hadn't refreshed didn't see you'd replied